### PR TITLE
Fix SkipReasonTestCase to allow serialization and deserialization.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- With CSproj based test runners test cases are passed across application boundaries which require test cases to be serializable.

https://github.com/Microsoft/vstest/issues/291

### CODE REVIEW

File: src/Microsoft.AspNetCore.Testing/xunit/SkipReasonTestCase.cs [hunk 1]
```diff
@@ -132,9 +138,36 @@ public void Deserialize(IXunitSerializationInfo info)
...
+                IReflectionMethodInfo reflectionMethod = Method as IReflectionMethodInfo;
+                if (reflectionMethod != null)
+                {
+                    TestMethodArguments = reflectionMethod.MethodInfo.ResolveMethodArguments(TestMethodArguments);
+                }
+
+                if (Method.IsGenericMethodDefinition)
+                {
+                    _methodGenericTypes = Method.ResolveGenericTypes(TestMethodArguments);
+                    Method = Method.MakeGenericMethod(MethodGenericTypes);
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Testing/xunit/SkipReasonTestCase.cs]: nit: Use _methodGenericTypes here

**Comment by BrennanConroy** [comment]
🚢 :it:

**Comment by Eilon** [comment]
Is it possible to add a test for the serialization logic to make sure it can serialize/deserialize an object?

**Comment by NTaylorMullen** [comment]
> Is it possible to add a test for the serialization logic to make sure it can serialize/deserialize an object?

Talked offline. No need for it.

**Pull request merged**